Gems/PhysX/Code/Source/BaseColliderComponent.cpp

```diff
@@ -328,7 +328,10 @@ namespace PhysX
             return false;
         }
 
-        const bool hasNonUniformScale = (AZ::NonUniformScaleRequestBus::FindFirstHandler(GetEntityId()) != nullptr);
+        const bool isAssetScaleUniform =
+            AZ::IsClose(physicsAssetConfiguration.m_assetScale.GetX(), physicsAssetConfiguration.m_assetScale.GetY()) &&
+            AZ::IsClose(physicsAssetConfiguration.m_assetScale.GetX(), physicsAssetConfiguration.m_assetScale.GetZ());
```

Comment:
thought: It's a shame we don't have a nice util for this thinking about it (I had a look on the `Vector3` API and couldn't see anything). I realize it's likely not worth it for this change but I do wonder how often this 'IsUniform' check comes up (maybe @greerdv wrote a helper for this?) or even if we had something like 'ElementsEqual/Close' for vector types... just thinking out loud really 😅

Reply:
moraaar and I already discussed this earlier. I'm not aware of a utility for this off the top of my head. I don't think it actually comes up that often to be honest.

Reply:
As a general abstraction, adding some helpers to AZ Math's Transform to check uniform for various data types/patterns might be a good place to consider for a later support.

I had also noticed that @shawstar had done some uniform checks from vectors for an export tool they working on, so there is definitely some potential need for this.

Reply:
Understood, my thought was a few function somewhere (not a member) but agree it's not always easy to know where it should go (though I'm sure we probably do have a MathsHelpers file somewhere). It's fine for now but this is the sort of thing that is definitely useful to add (and test! 🙂)